src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -105,10 +106,37 @@ def metadata_call_credentials(metadata_plugin, name=None):
     A CallCredentials object for use in a GRPCCallOptions object.
   """
   if name is None:
-    name = metadata_plugin.__name__
```

Comment:
@soltanmm 
As far as I can tell, objects having names is the exception and not the rule.  Is that the behavior we want, or are we just obfuscating naming unnecessarily?